libs/go/libp2p_node/aea/api.go

```diff
@@ -346,7 +346,9 @@ func (aea *AeaApi) Connect() error {
 	aea.closing = false
 	//TOFIX(LR) trade-offs between bufferd vs unbuffered channel
 	aea.out_queue = make(chan *Envelope, 10)
-	go aea.listen_for_envelopes()
+	aea.send_queue = make(chan *Envelope, 10)
+	go aea.listenForEnvelopes()
+	go aea.envelopeSendLoop()
```

Comment:
I think this is potentially problematic. Are we sure Envelopes cannot end up getting out of order?

Reply:
it's two independed loops, with status queue for sync, see no problem here.
send loop can be blocked, but receive will work anyway

Reply:
what guarantees status being consumed by right go routine? I see, the receive envelope loop puts into status channel

Reply:
thats the way it has to work